### PR TITLE
Add null record type

### PR DESCRIPTION
**Pull request by danni-m**

### CODE REVIEW

File: src/record.h [hunk 1]
```diff
@@ -76,14 +76,16 @@ extern RecordType* doubleRecordType;
...
 extern RecordType* asyncRecordType;
 extern Record StopRecord;
 extern Record WaitRecord;
 extern Record DummyRecord;
+extern Record NullRecord;
 
 Record* RG_GetDummyRecord();
 
-#define IS_SPECIAL_RECORD(r) (r == &StopRecord || r == &WaitRecord || r == &DummyRecord)
+#define IS_SPECIAL_RECORD(r) (r == &StopRecord || r == &WaitRecord || r == &DummyRecord || r == &NullRecord)
```
Comment (MeirShpilraien) [inline src/record.h:88]: I do not think NullRecord should be a special record...

File: src/record.c [hunk 2]
```diff
@@ -12,6 +12,7 @@ RecordType StopRecordType;
 Record StopRecord;
 Record WaitRecord;
 Record DummyRecord;
+Record NullRecord;
```
Comment (MeirShpilraien) [inline src/record.c:15]: I do not see where you initialize it, what promise in the if I will do GetType on NullRecord I will get nullRecordType?

**Review by MeirShpilraien — COMMENTED** [review]
:+1: Some comments

File: src/record.c (hunk 2 above)
Comment (danni-m) [reply src/record.c:15]: @MeirShpilraien its a static var, do we need to initialise it? if so how?

**Review by MeirShpilraien — APPROVED** [review]

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**